app/src/main/java/com/concordia/mcga/utilities/pathfinding/PathFinder.java

```diff
@@ -53,6 +56,41 @@ public PathFinder(TiledMap map) {
         return returnList;
     }
 
+    public List<PathFinderTile> shortestPathJunctions(int startX, int startY, int destX, int destY) throws Exception {
+        ArrayList<PathFinderTile> pathTiles = new ArrayList<PathFinderTile>(shortestPath(startX, startY, destX, destY));
+        ArrayList<PathFinderTile> pathTilesJunctions = new ArrayList<PathFinderTile>();
+
+        int curX = 0;
+        int curY = 0;
+        for (int i = 0; i < pathTiles.size(); i++) {
+            if (i == 0) {
+                PathFinderTile pft = pathTiles.get(i);
+                pathTilesJunctions.add(pft);
+                curX = pft.getCoordinateX();
+                curY = pft.getCoordinateY();
+            } else {
+                PathFinderTile pft = pathTiles.get(i);
+                if (!(pft.getCoordinateX() == curX && pft.getCoordinateY() != curY) && !(pft.getCoordinateY() == curY && pft.getCoordinateX() != curX)) {
```

Comment:
Equivalent expression it's a negated XOR

`if(!(pft.getCoordinateX()==curX ^ pft.getCoordinateY() == curY))`

DID YOU MEAN ? 

True when both XYcoordinates are the same or when both XYcoordinates are different, and false when X coincides but not Y and false when Y coincides but not X ?

I didn't run it but it sure does look 🐠